src/App.js

```diff
@@ -1,75 +1,205 @@
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todos: [],
+    // todosCopy: [],
+    title: '',
+    counter: 4,
+    filter: 'All',
+  }
+
+  onInputChange = (event) => {
+    this.setState({
+      title: event.target.value,
+    });
+  }
+
+  addTodo = (event) => {
+    const newitem = {
+      id: this.state.counter,
+      title: this.state.title,
+      completed: false,
+    };
+
+    if (event.key === 'Enter' && this.state.title) {
+      this.setState(prevState => ({
+        todos: [...prevState.todos, newitem],
+        counter: prevState.counter + 1,
+        title: '',
+        // todosCopy: [...prevState.todos, newitem],
+      }));
+    }
+  }
+
+  checkboxChange = (id) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((todo) => {
+        if (id !== todo.id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }),
+      // todosCopy: prevState.todos,
+    }));
+  }
+
+  deleteTodo = (id) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.filter(todo => id !== todo.id),
+      // todosCopy: prevState.todos,
+    }));
+  }
+
+  togleCompleted = () => {
+    if (this.state.todos.filter(todo => !todo.completed).length > 0) {
+      this.setState(prevState => ({
+        todos: prevState.todos.map(todo => ({
+          ...todo,
+          completed: true,
+        })),
+      }));
+    } else {
+      this.setState(prevState => ({
+        todos: prevState.todos.map(todo => ({
+          ...todo,
+          completed: false,
+        })),
+      }));
+    }
+  }
+
+  // showAll = () => {
+  //   this.setState(prevState => ({
+  //     todosCopy: prevState.todos.filter(todo => true),
+  //   }));
+  // }
+
+  // showActive = () => {
+  //   this.setState(prevState => ({
+  //     todosCopy: prevState.todos.filter(todo => !todo.completed),
+  //   }));
+  // }
+
+  // showCompleted = () => {
+  //   this.setState(prevState => ({
+  //     todosCopy: prevState.todos.filter(todo => todo.completed),
+  //   }));
+  // }
+
+  filterHandler = (paramtr) => {
+    this.setState({
+      filter: paramtr,
+    });
+  }
+
+  todosFilter = (item) => {
+    if (this.state.filter === 'Active') {
```

Comment:
Создай объект со значениями фильтров
```
const FILTER_TYPES = { 
  all: 'all',
  active: 'active',
  completed: 'completed',
}
```
```suggestion
    if (this.state.filter === FILTER_TYPES.active) {
```
Это позволит тебе, если что, поменять значение в одном месте, и оно поменяется везде, где ты его используешь. Плюс это гарантирует отсутствие ошибок из-за опечатки.